src/utils/run-status.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Fetch pipeline run events from the API endpoint
+ * @returns {Promise<Object>} Pipeline run events data in structured format
+ */
+export const fetchRunEvents = async () => {
+  try {
+    const response = await fetch(`/api/run-events`);
+    if (!response.ok) {
+      throw new Error(`Error fetching run events: ${response.statusText}`);
+    }
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error('Failed to load run events:', error);
+    return { nodes: {}, datasets: {}, pipeline: {} };
+  }
+};
+
+/**
+ * Process structured run events into a format usable by the flowchart
```

Comment:
hmm but then down here the description mentioned "usable by the flowchart"...I'm not very sure what this `processRunEvents` prefers to?

Reply:
Hey, my comment is about the function name processRunEvents and its description that says a format usable by the flowchart. I think both should be updated to better reflect their actual purpose